dashboard/src/components/common/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,146 @@
+import Input from '@/components/ui/v2/Input';
+import Button from '@/ui/v2/Button';
+import ChevronLeftIcon from '@/ui/v2/icons/ChevronLeftIcon';
+import ChevronRightIcon from '@/ui/v2/icons/ChevronRightIcon';
+import Text from '@/ui/v2/Text';
+import type { ButtonProps } from '@mui/material';
+import type { DetailedHTMLProps, HTMLProps } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+export type PaginationProps = DetailedHTMLProps<
+  HTMLProps<HTMLDivElement>,
+  HTMLDivElement
+> & {
+  /**
+   * Total number of pages.
+   */
+  totalNrOfPages: number;
+  elementsPerPage?: number;
+  /**
+   * Current page number.
+   */
+  currentPageNumber: number;
+  /**
+   * Function to be called when navigating to the previous page.
+   */
+  onPrevPageClick: VoidFunction;
+  /**
+   * Function to be called when navigating to the next page.
+   */
+  onNextPageClick: VoidFunction;
+  /**
+   * Props to be passed to the previous button component.
+   */
+  onChangePage: (page: number) => void;
```

Comment:
```suggestion
  onPageChange: (page: number) => void;
```

You'll need to change references too.